torch_geometric/nn/models/dimenet.py

```diff
@@ -59,7 +59,9 @@ def __init__(self, num_radial, cutoff=5.0, envelope_exponent=5):
         self.reset_parameters()
 
     def reset_parameters(self):
-        torch.arange(1, self.freq.numel() + 1, out=self.freq).mul_(PI)
+        with torch.no_grad():
```

Comment:
Do we need the `requires_grad_()` call here? IMO, it should be sufficient to just do:
```python
@torch.no_grad()
def reset_parameters(self):
```

Reply:
I am not sure about it, one question I have is that if we don't use `requires_grad_()` how do we track gradient for `self.freq`?

Reply:
I am not completely sure about it, I tried the following code snippets:
Snippet 1, with torch.no_grad:
```
PI = 3.14
class BBL1(torch.nn.Module):
    def __init__(self, num_radial, cutoff=5.0, envelope_exponent=5):
        super().__init__()
        self.freq = torch.nn.Parameter(torch.Tensor(num_radial))
        self.reset_parameters()

    @torch.no_grad()
    def reset_parameters(self):
        torch.arange(1, self.freq.numel() + 1, out=self.freq).mul_(PI)

    def forward(self, dist):
        raise NotImplementError

bbl1 = BBL1(4)
bbl1.freq.requires_grad  # False
```
The output was `False`. Snippet 2, with require_grad
```
class BBL2(torch.nn.Module):
    def __init__(self, num_radial, cutoff=5.0, envelope_exponent=5):
        super().__init__()
        self.freq = torch.nn.Parameter(torch.Tensor(num_radial))
        self.reset_parameters()

    def reset_parameters(self):
        with torch.no_grad():
            torch.arange(1, self.freq.numel() + 1, out=self.freq).mul_(PI)
        self.freq.requires_grad_()

    def forward(self, dist):
        raise NotImplementError

bbl2 = BBL2(4)
bbl2.freq.requires_grad  # True
```
The output was `True.`

Each returns a different output. Am I missing something here?

Reply:
Thanks!